services/collectibles/api.go

```diff
@@ -109,12 +111,43 @@ func (api *API) Deploy(ctx context.Context, chainID uint64, deploymentParameters
 	return DeploymentDetails{address.Hex(), tx.Hash().Hex()}, nil
 }
 
+func (api *API) DeployAssets(ctx context.Context, chainID uint64, deploymentParameters DeploymentParameters, txArgs transactions.SendTxArgs, password string) (DeploymentDetails, error) {
+
+	err := deploymentParameters.Validate()
+	if err != nil {
+		return DeploymentDetails{}, err
+	}
+
+	transactOpts := txArgs.ToTransactOpts(utils.GetSigner(chainID, api.accountsManager, api.config.KeyStoreDir, txArgs.From, password))
+
+	ethClient, err := api.RPCClient.EthClient(chainID)
+	if err != nil {
+		log.Error(err.Error())
+		return DeploymentDetails{}, err
+	}
+
+	address, tx, _, err := assets.DeployAssets(transactOpts, ethClient, deploymentParameters.Name,
+		deploymentParameters.Symbol, deploymentParameters.GetSupply())
+	if err != nil {
+		log.Error(err.Error())
+		return DeploymentDetails{}, err
+	}
+
+	return DeploymentDetails{address.Hex(), tx.Hash().Hex()}, nil
+}
+
 // Returns gas units + 10%
 func (api *API) DeployCollectiblesEstimate(ctx context.Context) (uint64, error) {
 	gasAmount := uint64(1960645)
 	return gasAmount + uint64(float32(gasAmount)*0.1), nil
 }
 
+// Returns gas units + 10%
+func (api *API) DeployAssetsEstimate(ctx context.Context) (uint64, error) {
+	gasAmount := uint64(957483)
```

Comment:
Where does this number come from?
Can we be sure it's constant across different chains?

Reply:
It is computed by hardhat deployment testing script.
Yes, it is the same on all chains.

Reply:
Okay, I think there's a tiny percentage of a chance that L2 clients can implement different gas costs for opcodes at which point this function could estimate incorrectly.

But that's an optimization we can handle if we run into that issue.